<div style="background:linear-gradient(135deg,#1e1b4b 0%,#4338ca 55%,#6366f1 100%);border-radius:18px;padding:34px 30px;color:#fff;font-family:Inter,Segoe UI,sans-serif">
  <div style="font-size:12px;letter-spacing:3px;color:#c7d2fe;font-weight:700;text-transform:uppercase">Chapter 32 · Solutions</div>
  <div style="font-size:34px;font-weight:900;line-height:1.1;margin:10px 0 6px">Counting Challenges, Worked Answers ✅</div>
  <div style="font-size:15px;color:#e0e7ff;max-width:720px;line-height:1.6">Full solutions to the five Counting &amp; Combinatorics challenges. Try them yourself first, then compare.</div>
  <div style="margin-top:16px;font-size:13px;color:#cbd5e1">Statistics, Data Science and AI: A Visual Handbook · John Fisher · 2026</div>
</div>

## ⚙️ Setup

In [1]:
import itertools
from math import factorial, perm, comb
print("ready")

ready


<div style="background:#eef2ff;border-left:5px solid #4f46e5;border-radius:10px;padding:16px 20px;font-family:Inter,sans-serif">
<span style="font-size:13px;font-weight:800;color:#4f46e5;letter-spacing:1px">CHALLENGE 1 · MULTIPLICATION</span>
<div style="font-size:21px;font-weight:800;color:#1a2138;margin-top:4px">Count the menus</div>
<div style="color:#4a5578;margin-top:6px">A cafe offers 4 mains, 3 sides, and 5 drinks. How many distinct meals (one of each)? What if you also pick one of 2 desserts?</div>
</div>

In [2]:
meals = 4 * 3 * 5
with_dessert = 4 * 3 * 5 * 2
print(f"meals = 4 x 3 x 5 = {meals}")
print(f"with a dessert = 4 x 3 x 5 x 2 = {with_dessert}")

meals = 4 x 3 x 5 = 60
with a dessert = 4 x 3 x 5 x 2 = 120


**Answer:** each independent choice multiplies, so 4 x 3 x 5 = 60 meals, and adding a 2-way dessert choice doubles it to 120. The multiplication principle scales effortlessly: every new decision just multiplies in its own number of options.

<div style="background:#eef2ff;border-left:5px solid #4f46e5;border-radius:10px;padding:16px 20px;font-family:Inter,sans-serif">
<span style="font-size:13px;font-weight:800;color:#2563eb;letter-spacing:1px">CHALLENGE 2 · PERMUTATIONS</span>
<div style="font-size:21px;font-weight:800;color:#1a2138;margin-top:4px">Order matters</div>
<div style="color:#4a5578;margin-top:6px">From a club of 10 members, in how many ways can you choose a President, a Vice-President, and a Treasurer (distinct roles)?</div>
</div>

In [3]:
n, r = 10, 3
print(f"P({n},{r}) = {n}!/({n}-{r})! = {perm(n,r)}")
print("check by enumeration:", len(list(itertools.permutations(range(n), r))))

P(10,3) = 10!/(10-3)! = 720
check by enumeration: 720


**Answer:** the three roles are different, so order matters and this is a permutation: P(10,3) = 10 x 9 x 8 = 720. (The president has 10 candidates, then 9 remain for VP, then 8 for treasurer.) If the three positions were interchangeable, it would instead be a combination.

<div style="background:#eef2ff;border-left:5px solid #4f46e5;border-radius:10px;padding:16px 20px;font-family:Inter,sans-serif">
<span style="font-size:13px;font-weight:800;color:#059669;letter-spacing:1px">CHALLENGE 3 · COMBINATIONS</span>
<div style="font-size:21px;font-weight:800;color:#1a2138;margin-top:4px">Order ignored</div>
<div style="color:#4a5578;margin-top:6px">From the same 10 members, how many ways to choose a 3-person committee where all roles are equal? How many handshakes occur if all 10 shake hands once?</div>
</div>

In [4]:
print(f"committees = C(10,3) = {comb(10,3)}")
print(f"handshakes = C(10,2) = {comb(10,2)}   (each handshake is a pair)")

committees = C(10,3) = 120
handshakes = C(10,2) = 45   (each handshake is a pair)


**Answer:** a committee has no internal order, so it is a combination: C(10,3) = 120, exactly the permutation 720 divided by 3! = 6. Handshakes are unordered pairs, so C(10,2) = 45. Whenever you are selecting a group rather than arranging one, reach for the combination.

<div style="background:#eef2ff;border-left:5px solid #4f46e5;border-radius:10px;padding:16px 20px;font-family:Inter,sans-serif">
<span style="font-size:13px;font-weight:800;color:#db2777;letter-spacing:1px">CHALLENGE 4 · WHICH ONE?</span>
<div style="font-size:21px;font-weight:800;color:#1a2138;margin-top:4px">Permutation or combination?</div>
<div style="color:#4a5578;margin-top:6px">For each, decide whether order matters and compute it: (a) a 4-character password from 26 letters with no repeats; (b) choosing 4 pizza toppings from 12.</div>
</div>

In [5]:
pw = perm(26,4)        # order matters: ABCD differs from DCBA
pizza = comb(12,4)     # order does not matter on a pizza
print(f"(a) password, order MATTERS  -> P(26,4) = {pw:,}")
print(f"(b) toppings, order IGNORED  -> C(12,4) = {pizza}")

(a) password, order MATTERS  -> P(26,4) = 358,800
(b) toppings, order IGNORED  -> C(12,4) = 495


**Answer:** (a) a password is ordered, ABCD is not DCBA, so it is a permutation: P(26,4) = 358,800. (b) pizza toppings are an unordered set, so it is a combination: C(12,4) = 495. The deciding question is always the same: would reordering the same items count as a different outcome?

<div style="background:#eef2ff;border-left:5px solid #4f46e5;border-radius:10px;padding:16px 20px;font-family:Inter,sans-serif">
<span style="font-size:13px;font-weight:800;color:#d97706;letter-spacing:1px">CHALLENGE 5 · COUNTING FOR PROBABILITY</span>
<div style="font-size:21px;font-weight:800;color:#1a2138;margin-top:4px">Favorable over total</div>
<div style="color:#4a5578;margin-top:6px">From a standard 52-card deck, what is the probability that a 5-card hand is exactly four-of-a-kind (four cards of one rank plus any fifth card)?</div>
</div>

In [6]:
hands = comb(52,5)
# 13 ranks for the quad, then the 5th card is any of the remaining 48
four_kind = 13 * comb(48,1)
p = four_kind / hands
print(f"total hands      = C(52,5) = {hands:,}")
print(f"four-of-a-kind   = 13 x 48 = {four_kind:,}")
print(f"P(four-of-a-kind)= {four_kind}/{hands} = {p:.8f}  (about 1 in {round(1/p):,})")

total hands      = C(52,5) = 2,598,960
four-of-a-kind   = 13 x 48 = 624
P(four-of-a-kind)= 624/2598960 = 0.00024010  (about 1 in 4,165)


**Answer:** count the favorable hands, choose the rank for the four-of-a-kind (13 ways), then any of the 48 remaining cards as the fifth, giving 13 x 48 = 624 hands out of C(52,5) = 2,598,960. So P ≈ 0.00024, about 1 in 4,165. Counting the favorable and total outcomes is the whole job; the probability is their ratio.

---
<div style="background:#ffffff;border:1px solid #e6e9f2;border-radius:16px;padding:22px 26px;font-family:Inter,sans-serif">
<div style="font-size:19px;font-weight:800;color:#1a2138">🎉 Nicely done</div>
<div style="color:#4a5578;line-height:1.8;margin-top:8px">You multiplied independent choices, told permutations from combinations by asking whether order matters, and counted favorable-over-total to turn counting into probability. These are the exact tools the binomial distribution is built from.</div>
</div>

<div style="text-align:center;color:#8b94b3;font-size:12px;margin-top:16px">Statistics, Data Science and AI: A Visual Handbook · © 2026 John Fisher</div>